### DISCUSSION TITLE
Installation on Linux fails

### DISCUSSION THREAD

**Discussion by matesmoliere** [samschott/maestral#905] [Q&A]
Trying to install as advice I got this:

antonio@Kayesh:~$ python3 -m venv maestral-venv
antonio@Kayesh:~$ source maestral-venv/bin/activate
(maestral-venv) antonio@Kayesh:~$ python3 -m pip install --upgrade 'maestral[gui]'
Collecting maestral[gui]
  Downloading maestral-1.7.2-py3-none-any.whl (437 kB)
     |████████████████████████████████| 437 kB 2.7 MB/s 
Collecting Pyro5>=5.10
  Downloading Pyro5-5.14-py3-none-any.whl (79 kB)
     |████████████████████████████████| 79 kB 2.9 MB/s 
Collecting requests>=2.16.2
  Downloading requests-2.30.0-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 1.6 MB/s 
Collecting pathspec>=0.5.8
  Downloading pathspec-0.11.1-py3-none-any.whl (29 kB)
Collecting survey<4.0,>=3.4.3
  Downloading survey-3.4.3-py3-none-any.whl (24 kB)
Collecting importlib-metadata>=3.6
  Downloading importlib_metadata-6.6.0-py3-none-any.whl (22 kB)
Collecting packaging
  Downloading packaging-23.1-py3-none-any.whl (48 kB)
     |████████████████████████████████| 48 kB 4.6 MB/s 
Collecting click>=8.0.2
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     |████████████████████████████████| 96 kB 4.1 MB/s 
Collecting rich
  Downloading rich-13.3.5-py3-none-any.whl (238 kB)
     |████████████████████████████████| 238 kB 3.2 MB/s 
Collecting watchdog>=2.0.1
  Downloading watchdog-3.0.0-py3-none-manylinux2014_x86_64.whl (82 kB)
     |████████████████████████████████| 82 kB 170 kB/s 
Collecting typing-extensions
  Downloading typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Requirement already satisfied, skipping upgrade: setuptools in ./maestral-venv/lib/python3.8/site-packages (from maestral[gui]) (44.0.0)
Collecting dropbox<12.0,>=11.28.0
  Downloading dropbox-11.36.0-py3-none-any.whl (594 kB)
     |████████████████████████████████| 594 kB 2.1 MB/s 
Collecting fasteners>=0.15
  Downloading fasteners-0.18-py3-none-any.whl (18 kB)
Collecting desktop-notifier>=3.3.0
  Downloading desktop_notifier-3.5.3-py3-none-any.whl (31 kB)
Collecting keyrings.alt>=3.1.0
  Downloading keyrings.alt-4.2.0-py3-none-any.whl (20 kB)
Collecting keyring>=22
  Downloading keyring-23.13.1-py3-none-any.whl (37 kB)
Collecting maestral-qt>=1.7.1; sys_platform == "linux" and extra == "gui"
  Downloading maestral_qt-1.7.2-py3-none-any.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB 2.9 MB/s 
Collecting serpent>=1.41
  Downloading serpent-1.41-py3-none-any.whl (9.6 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 204 kB/s 
Collecting certifi>=2017.4.17
  Downloading certifi-2023.5.7-py3-none-any.whl (156 kB)
     |████████████████████████████████| 156 kB 4.5 MB/s 
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.1.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (195 kB)
     |████████████████████████████████| 195 kB 5.4 MB/s 
Collecting urllib3<3,>=1.21.1
  Downloading urllib3-2.0.2-py3-none-any.whl (123 kB)
     |████████████████████████████████| 123 kB 4.0 MB/s 
Collecting wrapio>=1.0.0
  Downloading wrapio-2.0.0-py3-none-any.whl (5.8 kB)
Collecting zipp>=0.5
  Downloading zipp-3.15.0-py3-none-any.whl (6.8 kB)
Collecting markdown-it-py<3.0.0,>=2.2.0
  Downloading markdown_it_py-2.2.0-py3-none-any.whl (84 kB)
     |████████████████████████████████| 84 kB 3.3 MB/s 
Collecting pygments<3.0.0,>=2.13.0
  Downloading Pygments-2.15.1-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 5.1 MB/s 
Collecting six>=1.12.0
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting stone>=2.*
  Downloading stone-3.3.1-py3-none-any.whl (162 kB)
     |████████████████████████████████| 162 kB 3.8 MB/s 
Collecting importlib-resources; python_version < "3.9"
  Downloading importlib_resources-5.12.0-py3-none-any.whl (36 kB)
Collecting dbus-next; sys_platform == "linux"
  Downloading dbus_next-0.2.3-py3-none-any.whl (57 kB)
     |████████████████████████████████| 57 kB 3.1 MB/s 
Collecting jaraco.classes
  Downloading jaraco.classes-3.2.3-py3-none-any.whl (6.0 kB)
Collecting SecretStorage>=3.2; sys_platform == "linux"
  Downloading SecretStorage-3.3.3-py3-none-any.whl (15 kB)
Collecting jeepney>=0.4.2; sys_platform == "linux"
  Downloading jeepney-0.8.0-py3-none-any.whl (48 kB)
     |████████████████████████████████| 48 kB 2.3 MB/s 
Collecting PyQt6
  Downloading PyQt6-6.5.0.tar.gz (1.0 MB)
     |████████████████████████████████| 1.0 MB 2.9 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /home/antonio/maestral-venv/bin/python3 /tmp/tmpbsesb4lz prepare_metadata_for_build_wheel /tmp/tmpf2mljt4o
         cwd: /tmp/pip-install-ii3jwd8y/PyQt6
    Complete output (36 lines):
    Querying qmake about your Qt installation...
    Traceback (most recent call last):
      File "/tmp/tmpbsesb4lz", line 126, in prepare_metadata_for_build_wheel
        hook = backend.prepare_metadata_for_build_wheel
    AttributeError: module 'sipbuild.api' has no attribute 'prepare_metadata_for_build_wheel'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/tmp/tmpbsesb4lz", line 280, in <module>
        main()
      File "/tmp/tmpbsesb4lz", line 263, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/tmp/tmpbsesb4lz", line 130, in prepare_metadata_for_build_wheel
        return _get_wheel_metadata_from_wheel(backend, metadata_directory,
      File "/tmp/tmpbsesb4lz", line 159, in _get_wheel_metadata_from_wheel
        whl_basename = backend.build_wheel(metadata_directory, config_settings)
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/sipbuild/api.py", line 46, in build_wheel
        project = AbstractProject.bootstrap('wheel',
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/sipbuild/abstract_project.py", line 87, in bootstrap
        project.setup(pyproject, tool, tool_description)
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/sipbuild/project.py", line 586, in setup
        self.apply_user_defaults(tool)
      File "/tmp/pip-install-ii3jwd8y/PyQt6/project.py", line 66, in apply_user_defaults
        super().apply_user_defaults(tool)
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/pyqtbuild/project.py", line 70, in apply_user_defaults
        super().apply_user_defaults(tool)
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/sipbuild/project.py", line 237, in apply_user_defaults
        self.builder.apply_user_defaults(tool)
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/pyqtbuild/builder.py", line 78, in apply_user_defaults
        self._get_qt_configuration()
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/pyqtbuild/builder.py", line 493, in _get_qt_configuration
        for line in project.read_command_pipe([self.qmake, '-query']):
      File "/tmp/pip-build-env-7sbmp_6g/overlay/lib/python3.8/site-packages/sipbuild/project.py", line 554, in read_command_pipe
        raise UserException(
    sipbuild.exceptions.UserException
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/antonio/maestral-venv/bin/python3 /tmp/tmpbsesb4lz prepare_metadata_for_build_wheel /tmp/tmpf2mljt4o Check the logs for full command output.
(maestral-venv) antonio@Kayesh:~$

**Comment by samschott** [comment]
Looks like its trying to build PyQt6 from source instead of downloading the wheels from PyPI. My best guess is that PyQt6 wheels are not available for your system / distribution.

Could you provide the information that I ask for in the bug template? Also potentially relevant here, do you have an ARM or x64 / Intel CPU?

Building Qt and PyQt from source is a tall order, I'm not surprised that it fails. A workaround for now would be to use the daemon without a GUI.
